Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs

```diff
@@ -71,12 +71,26 @@ public class ApplicationInsightsOptions
     public bool EnableClientSideTracking { get; set; } = true;
 
     /// <summary>
-    /// Gets or sets a value indicating whether local development is enabled where telemetry is sent to
-    /// Azure Application Insights from the local instance.
-    /// Will only work if local authentication is enabled in the properties of the Applications Insights resource
-    /// which should only be the case during development and testing.
+    /// Gets or sets a value indicating whether local development is enabled where telemetry is sent to a secure Azure
+    /// Application Insights resource from the local instance. When using this option you will also need to provide
+    /// tenantId, clientId, and clientSecret of the service principal.
     /// </summary>
-    public bool EnableLocalDevelopment { get; set; }
+    public bool EnableSecureLocalDevelopment { get; set; }
```

Comment:
To follow what AI does, I'd rather use:

```suggestion
    public bool EnableSecureControlChannelLocalAuthentication { get; set; }
```

And explain in the comment, with a link, what it is.

But! I'm confused about whether using the credentials of a Service Principal is considered Local Authentication or not (a Managed Identity definitely isn't). Here I assumed it is, but I'm not sure, please check.

Let's first finish the convo about this below: https://github.com/Lombiq/Orchard-Azure-Application-Insights/pull/85#discussion_r1628492342